modApiServer/src/org/aion/api/server/rpc/ApiWeb3Aion.java

```diff
@@ -2817,7 +2818,7 @@ private JSONObject buildResponse() {
 
         MinerStatsView update() {
             // get the latest head
-            Block blk = getBestBlock();
+            Block blk = getBestMiningBlock();
```

Comment:
We should force the call return AionBlock because it always returns a mining block

Reply:
Agreed but the blk variable must use the base type because the variable may also hold staking blocks.

Reply:
If you use the getBestMiningBlock call, the kernel should guarantee you always get the AionBlcok.So you need to modify the return of the ApiAion class.

Reply:
The blk variable is used as a temporary variable, like this: `blk = getBlockByHash(blk.getParentHash());`, which could return a pos block.